sdk/cosmos/azure_data_cosmos/src/models/connection_string.rs

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+use std::borrow::Cow;
+
+use azure_core::{credentials::Secret, fmt::SafeDebug, Error};
+use serde::Deserialize;
+
+/// Represents a Cosmos DB connection string.
+///
+/// The [`Debug`] implementation will not print the account key.
+#[derive(Clone, Deserialize, PartialEq, Eq, SafeDebug)]
```

Comment:
I don't think this needs to be `Deserialize`. It's not something we need to deserialize from JSON.

```suggestion
#[derive(Clone, PartialEq, Eq, SafeDebug)]
```